server/app/services/scheduled_nps_report_service.py

```
@@ -0,0 +1,347 @@
+from app.services.nps_report_service import NPSReportService
+from app.services.nps_email_service import NPSEmailService
+from app.services.pdf_report_service import PDFReportService
+
+logger = logging.getLogger(__name__)
+
+
+class ScheduledNPSReportService:
+    """Service for managing scheduled NPS reports."""
+
+    @staticmethod
+    def _calculate_next_scheduled_time(frequency: ReportFrequency, current_time: datetime = None) -> datetime:
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)